### DISCUSSION TITLE
Blocks Coding - select random number from a list of values... and gears

### DISCUSSION THREAD

**Discussion by rbeckett75** [pybricks/support#1340] [Q&A]
Now that I have the return to center working - my creation needs to set it at a random angle on each loop
I've got it working with the random block from -90 to +90, and it works great:

![rotate-random-working](https://github.com/pybricks/support/assets/90262685/186f3aa6-2b7b-4dc1-84e3-f1a34faceaf7)

However, with this creation, it might be better if define some angles, and get it to pick one of them each time - as in the first one, sometimes the random angle it picks is too small to really notice.  So I'd like to have a list, eg: -90, -45, -15, 0, 20, 50, 80

But.. it's not working - and I'm guessing it has to do with text/number differences?
in this example, it won't print anything if GET RANDOM is selected (name random isn't defined)

![datalist-randomisntdefined](https://github.com/pybricks/support/assets/90262685/a5920698-684b-4602-bb2e-b785428cea4f)

However it does work, if I select GET FIRST or LAST...

### Solution!
.....well.. while typing this question.. I managed to solve it.. So for the benefit of others - I'll show what worked! :)
I set it to pick a specific number.. and then added random from 0 to the total numbers in the list.

![solved!](https://github.com/pybricks/support/assets/90262685/33ddd58e-d038-4327-a090-3cd46eb65aa4)

### Gears?

Now.. since I can't see any way to do Gears in a Motor Block (I'm assuming it's a feature yet to come!)
![gears](https://github.com/pybricks/support/assets/90262685/9c94cea7-48d4-4c07-9103-33be54bfafb3)

while I'm testing, I run this bit of code, to see what the limits are, and then I update my list of values (with larger degrees)
![discoverlimits](https://github.com/pybricks/support/assets/90262685/645e65d8-44ba-4bf3-b7bb-a91f01e8a59b)
![manual-geared-limitfinder](https://github.com/pybricks/support/assets/90262685/885167f2-9fea-4c3c-9ceb-76e2cebd5b64)

and then I update my list of values (with larger degrees)
![biglimits](https://github.com/pybricks/support/assets/90262685/a294b9ff-847e-4b44-baf3-9ed83bdb2a11)

**Comment by laurensvalk** [comment]
If you want -/+90 but in increments, another simple approach might be to do something like:

(random from -6 to 6) * 15.

-------------------

I was on the fence about including `gears`. Since you're asking, I guess we just should :smile:

**Comment by rbeckett75** [comment]
While I haven't needed to use the gears in previous projects..  I could see how it would help in something like this..  as the angles would be 'real life' as opposed to multiple revolutions of the motor to get the same result

and expanding list of numbers (like you do with the list) would work - although having drop-down options and small images of each gear would be very cool!
